api/v1alpha1/basic_auth_types.go

```diff
@@ -26,4 +26,10 @@ type BasicAuth struct {
 	//
 	// Note: The secret must be in the same namespace as the SecurityPolicy.
 	Users gwapiv1.SecretObjectReference `json:"users"`
+
+	// This field specifies the header name to forward a successfully authenticated user to
+	// the backend. The header will be added to the request with the username as the value.
+	//
+	// If it is not specified, the username will not be forwarded.
+	ForwardUsernameHeader string `json:"forwardUsernameHeader,omitempty"`
```

Comment:
```suggestion
        //
        // +optional
	ForwardUsernameHeader *string `json:"forwardUsernameHeader,omitempty"`
```

this should be opt-in.

Reply:
why change this in implement PR?